components/patient-access-rules/migrations/src/main/java/org/phenotips/data/permissions/internal/R71502PhenoTips3226DataMigration.java

```diff
@@ -0,0 +1,142 @@
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.store.XWikiHibernateBaseStore;
+import com.xpn.xwiki.store.XWikiHibernateStore;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.AbstractHibernateDataMigration;
+import com.xpn.xwiki.store.migration.hibernate.HibernateDataMigration;
+
+/**
+ * Migration for PhenoTips issue #3226: Search for documents with patients that have XWikiRights objects and remove the
+ * objects.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Component(roles = { HibernateDataMigration.class })
+@Named("71502-PT-3226")
+@Singleton
+public class R71502PhenoTips3226DataMigration extends AbstractHibernateDataMigration implements
+    XWikiHibernateBaseStore.HibernateCallback<Object>
+{
+    /** XWiki class that contains rights to XWiki documents. */
+    private static final EntityReference RIGHTS_CLASS =
+        new EntityReference("XWikiRights", EntityType.DOCUMENT, new EntityReference("XWiki", EntityType.SPACE));
+
+    /** Resolves class names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<EntityReference> entityResolver;
+
+    /** Logging helper object. */
+    @Inject
+    private Logger logger;
+
+    /** Serializes the owner name. */
+    @Inject
+    private EntityReferenceSerializer<String> serializer;
+
+    /** Resolves unprefixed document names to the current wiki. */
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> resolver;
+
+    @Override protected void hibernateMigrate() throws DataMigrationException, XWikiException
+    {
+        getStore().executeWrite(getXWikiContext(), this);
+    }
+
+    @Override public String getDescription()
+    {
+        return "Search for documents with patients that have XWikiRights objects and remove the objects.";
+    }
+
+    @Override public XWikiDBVersion getVersion()
+    {
+        return new XWikiDBVersion(71502);
+    }
+
+    /**
+     * Searches for all patient documents that contain a XWikiRights object, and removes the objects
+     */
+    @Override
+    public Object doInHibernate(final Session session) throws HibernateException, XWikiException
+    {
+        XWikiContext context = getXWikiContext();
+        XWiki xwiki = context.getWiki();
+        final DocumentReference rightsClassReference = this.entityResolver.resolve(RIGHTS_CLASS);
+        // FIXME this query does not generate results
+        Query q =
+            session.createQuery("select distinct doc.fullName from XWikiDocument as doc, BaseObject as p, BaseObject as r where "
+                + "p.name = doc.fullName and p.className = '" + this.serializer.serialize(Patient.CLASS_REFERENCE)
+                + "' and r.name = doc.fullName and r.className = '" + this.serializer.serialize(rightsClassReference) + "'");
```

Comment:
@sdumitriu It looks like this query is not retrieving any documents for some reason, but it looks okay to me. Do you see anything that may be an issue here?